gradoop-flink/src/main/java/org/gradoop/flink/model/api/operators/BaseGraphCollectionOperatorSupport.java

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright © 2014 - 2019 Leipzig University (Database Research Group)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.gradoop.flink.model.api.operators;
+
+import org.gradoop.flink.model.api.epgm.BaseGraph;
+import org.gradoop.flink.model.api.epgm.BaseGraphCollection;
+import org.gradoop.flink.model.impl.operators.combination.Combination;
+import org.gradoop.flink.model.impl.operators.exclusion.Exclusion;
+import org.gradoop.flink.model.impl.operators.overlap.Overlap;
+
+/**
+ * Defines callFor methods for graph collections.
+ *
+ * @param <LG> graph type
+ * @param <GC> graph collection type
+ */
+public interface BaseGraphCollectionOperatorSupport<LG extends BaseGraph, GC extends BaseGraphCollection> {
+
+  /**
+   * Creates a value using the given unary collection to value operator.
+   *
+   * @param operator unary graph collection to value operator
+   * @param <T> return type
+   * @return result of given operator
+   */
+  <T> T callForValue(UnaryBaseGraphCollectionToValueOperator<GC, T> operator);
+
+  /**
+   * Calls the given binary collection to value operator using that graph collection and the
```

Comment:
Maybe
```suggestion
   * Calls the given binary collection to value operator using this graph collection and the
```
since the graph collection implementing this interface is meant here